src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int COUNT_OF_FIGURE = 6;
+
+    public static void main(String[] args) {
+
+        FigureSupplier figureSuplier = new FigureSupplier();
+        Figure[] allFigures = new Figure[COUNT_OF_FIGURE];
+        for (int i = 0; i < allFigures.length; i++) {
+            if (i < allFigures.length / 2) {
+                allFigures[i] = figureSuplier.getRandomFigure();
+            } else {
+                allFigures[i] = figureSuplier.getDefaultFigure();
+            }
+            allFigures[i].drawFigure();
+        }
```

Comment:
You should create only one instance of `FigureSupplier` before the loop starts, which is done correctly here. However, make sure not to instantiate `FigureSupplier` within the loop, as that would be inefficient and against the checklist guidelines.